src/Serializer/NormalizationHelper.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace eLife\ApiSdk\Serializer;
+
+use eLife\ApiClient\Result;
+use eLife\ApiSdk\Collection\ArraySequence;
+use eLife\ApiSdk\Collection\Sequence;
+use GuzzleHttp\Promise\PromiseInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * A better name is welcome.
+ */
+final class NormalizationHelper
+{
+    /**
+     * @var NormalizerInterface
+     */
+    private $normalizer;
+
+    /**
+     * @var DenormalizerInterface
+     */
+    private $denormalizer;
+
+    /**
+     * @var string|null
+     */
+    private $format;
+
+    /**
+     * @param string|null $format
+     */
+    public function __construct(NormalizerInterface $normalizer, DenormalizerInterface $denormalizer, $format)
+    {
+        $this->normalizer = $normalizer;
+        $this->denormalizer = $denormalizer;
+        $this->format = $format;
+    }
+
+    public function selectField(PromiseInterface $resultPromise, $fieldName, $default = null) : PromiseInterface
+    {
+        return $resultPromise->then(function (Result $entity) use ($fieldName, $default) {
+            if ($default !== null) {
+                return $entity[$fieldName] ?? $default;
+            } else {
+                return $entity[$fieldName];
+            }
+        });
+    }
+
+    public function denormalizePromise(PromiseInterface $promise, string $class, $context) : PromiseInterface
+    {
+        return $promise->then(function (array $entity) use ($class, $context) {
+            return $this->denormalizer->denormalize($entity, $class, $this->format, $context);
+        });
+    }
+
+    public function denormalizeSequence(Sequence $sequence, string $class, $context) : Sequence
+    {
+        return $sequence->map(function (array $entity) use ($class, $context) {
+            return $this->denormalizer->denormalize($entity, $class, $this->format, $context);
+        });
+    }
+
+    public function denormalizeArray(array $array, string $class, $context) : ArraySequence
+    {
+        return new ArraySequence(array_map(function (array $subject) use ($class, $context) {
+            return $this->denormalizer->denormalize($subject, $class, $this->format, $context);
+        }, $array));
+    }
+
+    public function normalizeSequenceToSnippets(Sequence $sequence, $context) : array
+    {
+        return $sequence->map(function ($each) use ($context) {
+            $context['snippet'] = true;
+
+            return $this->normalizer->normalize($each, $this->format, $context);
```

Comment:
Is there anything else in `$context` besides `snippet`? If not, we can use `normalizeToSnippet()` in the `map()` and also remove a `$context` parameter.